pkg/router/template/types.go

```diff
@@ -136,6 +140,88 @@ type certificateWriter interface {
 	DeleteCertificate(directory, id string) error
 }
 
+// ConfigManagerOptions is the options passed to a template router's
+// configuration manager.
+type ConfigManagerOptions struct {
+	// ConnectionInfo specifies how to connect to the underlying router.
+	ConnectionInfo string
+
+	// CommitInterval specifies how often to commit changes made to the
+	// underlying router via the configuration manager.
+	CommitInterval time.Duration
+
+	// BlueprintRoutes are a list of routes blueprints pre-allocated by
+	// the config manager to dynamically manage route additions.
+	BlueprintRoutes []*routeapi.Route
+
+	// BlueprintRoutePoolSize is the size of the pre-allocated pool for
+	// each route blueprint. This can be overriden on an individual
+	// route basis with a route annotation:
+	//    router.openshift.io/pool-size
+	BlueprintRoutePoolSize int
+
+	// DynamicServerPrefix is the prefix used for naming the dynamic
+	// servers associated with a route. These dynamic servers are used to
+	// quickly modify the router config for any endpoint changes.
+	DynamicServerPrefix string
+
+	// MaxDynamicServers is the maximum number of dynamic servers we
+	// will allocate on a per-route basis.
+	MaxDynamicServers int
+}
+
+// ConfigManager is used by the router to make configuration changes using
+// the template router's dynamic configuration API (if any).
+type ConfigManager interface {
```

Comment:
Please specify the threading guarantees this code requires with the router controller here.

Reply:
ack